src/main/scala/edu/ie3/simona/agent/grid/GridResultsSupport.scala

```diff
@@ -542,15 +542,52 @@ private[grid] trait GridResultsSupport {
   ): (ComparableQuantity[ElectricCurrent], ComparableQuantity[Angle]) =
     (
       iNominal.multiply(iPu.abs).asComparable,
-      Quantities.getQuantity(
-        atan(iPu.imag / iPu.real).toDegrees + max(
-          0.0,
-          -signum(iPu.real)
-        ) * 180.0,
-        PowerSystemUnits.DEGREE_GEOM
-      )
+      angle(iPu)
     )
 
+  /** Correct the offset of an angle dependent on the direction. If the
+    * direction is negative, 180° are added
+    */
```

Comment:
Okay so I had to draw that stuff to see why what we are doing here makes sense so I would extend the documentation a little bit.

To me what we are doing seems to be to adjust an angle so instead of getting the relative angle we get its angle with respect to the real axis. This means that in case of a negative real value we need to offset it by + 180°

Reply:
I also just had to quickly sketch this out :D 
I refactored the code a little bit and added commentary, lmk what you think